### ISSUE TITLE
[BUG]CPack: The -DX_VCPKG_APPLOCAL_DEPS_INSTALL=ON is invalid

### ISSUE THREAD

**Issue by KangLin** [microsoft/vcpkg#32560]
**CPack: The -DX_VCPKG_APPLOCAL_DEPS_INSTALL=ON is invalid**

ag: vcpkg is installed openssl

```
vcpkg install openssl
```

Use the CMAKE install command to install dependent libraries(libcrypto-3-x64.dll,libssl-3-x64.dll).

```
git clone https://github.com/KangLin/RabbitCommon.git
mkdir build
cd build
cmake ..  -DCMAKE_INSTALL_PREFIX=install  -DCMAKE_BUILD_TYPE=Release ^
              -DCMAKE_TOOLCHAIN_FILE=${{env.VCPKG_ROOT}}/scripts/buildsystems/vcpkg.cmake ^
              -DVCPKG_VERBOSE=ON ^
              -DX_VCPKG_APPLOCAL_DEPS_INSTALL=ON
cmake --build . --config Release --target install
```

But use the **CPack** command to is not install dependent libraries(libcrypto-3-x64.dll,libssl-3-x64.dll).

```
git clone https://github.com/KangLin/RabbitCommon.git
mkdir build
cd build
cmake ..  -DCMAKE_INSTALL_PREFIX=install  -DCMAKE_BUILD_TYPE=Release ^
              -DCMAKE_TOOLCHAIN_FILE=${{env.VCPKG_ROOT}}/scripts/buildsystems/vcpkg.cmake ^
              -DVCPKG_VERBOSE=ON ^
              -DX_VCPKG_APPLOCAL_DEPS_INSTALL=ON
cmake --build . --config Release --target package
```

**Environment**
- OS: [Windows]
- Compiler: MSVC

@autoantwort @strega-nil @sandercox @crud89 @dhrdlicka @Be-ing

**Comment by Osyotr** [comment]
Is your application linked against openssl? If not (i.e. you are using LoadLibrary to load it), then you have to deploy it manually.

**Comment by KangLin** [comment]
@Osyotr 　Has used FIND_PACKAGE(openssl) and link it in my project.

**Comment by dg0yt** [comment]
> Has used FIND_PACKAGE(openssl) and link it in my project.

I can't find it in https://github.com/KangLin/ChineseChessControl.

**Comment by KangLin** [comment]
@dg0yt In https://github.com/KangLin/RabbitCommon

**Comment by KangLin** [comment]
@LilyWangLL @dg0yt @Osyotr @autoantwort @strega-nil @sandercox @crud89 @dhrdlicka @Be-ing

I wrote a simple project for the new.
 Discovery: Requires
- find_package(OpenSSL)
- target_link_libraries(test PRIVATE OpenSSL::SSL OpenSSL::Crypto)
- need to use libCrypto API in source code

It works fine.

When the COMPONENT is same. it works fine. See: test_ok\encrypt\CMakeLists.txt
```
install(TARGETS ${PROJECT_NAME}
    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
        COMPONENT Runtime
    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
        COMPONENT Runtime
    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
        COMPONENT Runtime
)

```

Ｗhen the COMPONENT is diffent. it is not works. See: test\encrypt\CMakeLists.txt
```
install(TARGETS ${PROJECT_NAME}
    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
        COMPONENT Runtime
    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
        COMPONENT Runtime
    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
        COMPONENT Development
)
```


Please see test is not work. test_ok is works fine.

[test.zip](https://github.com/microsoft/vcpkg/files/12075268/test.zip)

**Comment by sandercox** [comment]
I tried to do this but cannot see what you mean doesn't work.

Can you explain what you would have expected to happen what files to go where and what is not working. For me both examples output the same...

```
install/bin/encrypt.dll
install/bin/libcrypto-3-x64.dll
install/bin/test.exe

install/lib/encrypt.lib
```

**Comment by KangLin** [comment]
@sandercox 

- Please use :
```
cmake --build . --config Release --target package
```

- expected: install libcrypto-3-x64.dll
- results:
Don't find libcrypto-3-x64.dll in  D:\\source\\temp\\test\\build\\_CPack_Packages\\win64\\NSIS\\test-0.1.1-win64
So that don't find libcrypto-3-x64.dll in D:/source/temp/test/build/test-0.1.1-win64.exe


- The detail:

D:\source\temp\test>mkdir build

D:\source\temp\test>cd build

**D:\source\temp\test\build>cmake .. -DCMAKE_BUILD_TYPE=Release -DX_VCPKG_APPLOCAL_DEPS_INSTALL=ON -DCMAKE_TOOLCHAIN_FILE=d:/source/vcpkg/scripts/buildsystems/vcpkg.cmake**
-- Building for: Visual Studio 17 2022
-- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.19045.
-- The CXX compiler identification is MSVC 19.35.32216.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.35.32215/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenSSL: optimized;D:/source/vcpkg/installed/x64-windows/lib/libcrypto.lib;debug;D:/source/vcpkg/installed/x64-windows/debug/lib/libcrypto.lib (found version "3.1.1")
-- Configuring done
-- Generating done
-- Build files have been written to: D:/source/temp/test/build

**D:\source\temp\test\build>cmake --build . --config Release --target package**
MSBuild version 17.5.1+f6fdcf537 for .NET Framework

  Checking Build System
  Building Custom Rule D:/source/temp/test/encrypt/CMakeLists.txt
  encrypt.cpp
    正在创建库 D:/source/temp/test/build/encrypt/Release/encrypt.lib 和对象
   D:/source/temp/test/build/encrypt/Release/encrypt.exp
  encrypt.vcxproj -> D:\source\temp\test\build\encrypt\Release\encr
  ypt.dll
  Building Custom Rule D:/source/temp/test/CMakeLists.txt
  main.cpp
  test.vcxproj -> D:\source\temp\test\build\Release\test.exe
  Building Custom Rule D:/source/temp/test/CMakeLists.txt
  CPack: Create package using NSIS
  CPack: Install projects
  CPack: - Install project: test [Release]
  CPack: -   Install component: Development
  -- Installing app dependencies for encrypt...
  CPack: -   Install component: Runtime
  -- Installing app dependencies for test...
  CPack: Create package
  CPack: - package: D:/source/temp/test/build/test-0.1.1-win64.exe
  generated.
  
D:\source\temp\test\build>cd _CPack_Packages\win64\NSIS\test-0.1.1-win64\Development\lib

D:\source\temp\test\build\_CPack_Packages\win64\NSIS\test-0.1.1-win64\Development\lib>dir

 D:\source\temp\test\build\_CPack_Packages\win64\NSIS\test-0.1.1-win64\Development\lib 的目录

2023/07/21  08:33    <DIR>          .
2023/07/21  08:33    <DIR>          ..
2023/07/21  08:33             2,252 encrypt.lib
               1 个文件          2,252 字节
               2 个目录 146,222,755,840 可用字节

D:\source\temp\test\build\_CPack_Packages\win64\NSIS\test-0.1.1-win64\Runtime\bin>dir

 D:\source\temp\test\build\_CPack_Packages\win64\NSIS\test-0.1.1-win64\Runtime\bin 的目录

2023/07/21  08:33    <DIR>          .
2023/07/21  08:33    <DIR>          ..
2023/07/21  08:33            10,240 encrypt.dll
2023/07/21  08:33            11,776 test.exe
               2 个文件         22,016 字节
               2 个目录 146,222,755,840 可用字节

**Comment by sandercox** [comment]
Thank you for your explanation how to observer the issue. I can now reproduce.

We have to re-do the parsing of the install command to make this work and our current solution apparently doesn't catch all the possible methods. I think I have a fix for this (PR #32683) and otherwise I have a workaround :-)

Workaround would be to split your install command in two commands:
```cmake
install(TARGETS ${PROJECT_NAME}
    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
    COMPONENT Runtime
    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
    COMPONENT Runtime
)
install(TARGETS ${PROJECT_NAME}
    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
    COMPONENT Development
)
```
